whois-query/src/main/java/net/ripe/db/whois/query/pipeline/WhoisServerPipelineFactory.java

```diff
@@ -1,116 +0,0 @@
-package net.ripe.db.whois.query.pipeline;
-
-import net.ripe.db.whois.common.ApplicationVersion;
-import net.ripe.db.whois.common.pipeline.MaintenanceHandler;
-import net.ripe.db.whois.query.handler.QueryHandler;
-import org.jboss.netty.buffer.ChannelBuffer;
-import org.jboss.netty.buffer.ChannelBuffers;
-import org.jboss.netty.channel.ChannelPipeline;
-import org.jboss.netty.channel.ChannelPipelineFactory;
-import org.jboss.netty.channel.Channels;
-import org.jboss.netty.handler.codec.frame.DelimiterBasedFrameDecoder;
-import org.jboss.netty.handler.codec.string.StringDecoder;
-import org.jboss.netty.handler.execution.ExecutionHandler;
-import org.jboss.netty.handler.execution.OrderedMemoryAwareThreadPoolExecutor;
-import org.jboss.netty.handler.timeout.ReadTimeoutHandler;
-import org.jboss.netty.handler.timeout.WriteTimeoutHandler;
-import org.jboss.netty.util.HashedWheelTimer;
-import org.jboss.netty.util.Timer;
-import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.stereotype.Component;
-
-import javax.annotation.PreDestroy;
-import java.nio.charset.StandardCharsets;
-import java.util.concurrent.ThreadFactory;
-import java.util.concurrent.TimeUnit;
-import java.util.concurrent.atomic.AtomicInteger;
-
-@Component
-public class WhoisServerPipelineFactory implements ChannelPipelineFactory {
-
-    private static final ChannelBuffer LINE_DELIMITER = ChannelBuffers.wrappedBuffer(new byte[]{'\n'});
-    private static final ChannelBuffer INTERRUPT_DELIMITER = ChannelBuffers.wrappedBuffer(new byte[]{(byte)0xff, (byte)0xf4, (byte)0xff, (byte)0xfd, (byte)0x6});
-
-    private static final Timer TIMER = new HashedWheelTimer();
-    private static final int TIMEOUT_SECONDS = 180;
-    private static final int POOL_SIZE = 64;
-    private static final int MEMORY_SIZE_UNLIMITED = 0;
-
-    private final ReadTimeoutHandler readTimeoutHandler = new ReadTimeoutHandler(TIMER, TIMEOUT_SECONDS, TimeUnit.SECONDS);
-    private final WriteTimeoutHandler writeTimeoutHandler = new WriteTimeoutHandler(TIMER, TIMEOUT_SECONDS, TimeUnit.SECONDS);
-    private final StringDecoder stringDecoder = new StringDecoder(StandardCharsets.UTF_8);
-
-    private final ExecutionHandler executionHandler = new ExecutionHandler(new OrderedMemoryAwareThreadPoolExecutor(
```

Comment:
`ExecutionHandler` has been removed in Netty 4 and replaced with `EventExecutor`. See [No more ExecutionHandler - it's in the core](https://netty.io/wiki/new-and-noteworthy-in-4.0.html#no-more-executionhandler---its-in-the-core).

Depending on the result of the load testing, the following implementations of `ExecutionHandler` _might_ be configured and used:

1. [DefaultEventExecutor](https://netty.io/4.1/api/io/netty/util/concurrent/DefaultEventExecutor.html). This might require having to use those constructors which allows provision of implementation of [ThreadFactory](https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/ThreadFactory.html) and [Executor](https://docs.oracle.com/javase/10/docs/api/java/util/concurrent/Executor.html) 
2. [ImmediateEventExecutor](https://netty.io/4.1/api/io/netty/util/concurrent/ImmediateEventExecutor.html)

Reply:
I'm testing using an EventExecutorGroup to avoid blocking the Netty I/O thread.

See https://livebook.manning.com/book/netty-in-action/chapter-7/1